Sources/KituraNet/BufferList.swift

```diff
@@ -147,7 +147,12 @@ public class BufferList {
 
     */
     public func fill(array: inout [UInt8]) -> Int {
-        return fill(buffer: UnsafeMutablePointer(mutating: array), length: array.count)
+        var filledBytes = 0
+        let arrayCount = array.count
+        array.withUnsafeMutableBufferPointer({ bufferPointer in
+            filledBytes = fill(buffer: bufferPointer.baseAddress!, length: arrayCount)
+        })
+        return filledBytes
```

Comment:
btw, for the `fill` into an `UnsafeMutablePointer` (which should be deprecated), I'd suggest

```swift
    public func fill(buffer: UnsafeMutablePointer<UInt8>, length: Int) -> Int {
        // NOTE: This API should be deprecated and expressed as either `UnsafeMutableBufferPointer<UInt8>` or
        // preferrable `UnsafeMutableRawBufferPointer` or even better: replaced with nothing.

        let bytesToFill = min(length, self.byteBuffer.readableBytes)

        // Get a `ByteBufferView` (a `Collection`) of all readable bytes.
        let readableByteView = self.byteBuffer.readableBytesView

        // We want to overwrite the first `bytesToFill` bytes with the first `bytesToFill` bytes from the `ByteBufferView`.
        _ = UnsafeMutableBufferPointer(start: buffer, count: bytesToFill)
            .initialize(from: readableByteView.prefix(bytesToFill))

        // And then move the reader index to indicate that we consumed the bytes.
        self.byteBuffer.moveReaderIndex(forwardBy: bytesToFill)

        return bytesToFill
    }
```

Reply:
Thanks again!